extruct/__init__.py

```diff
@@ -1,16 +1,107 @@
+import logging
+import functools
 from lxml.html import fromstring
-
 from extruct.jsonld import JsonLdExtractor
 from extruct.rdfa import RDFaExtractor
 from extruct.w3cmicrodata import MicrodataExtractor
+from extruct.opengraph import OpenGraphExtractor
+from extruct.microformat import MicroformatExtractor
 from extruct.xmldom import XmlDomHTMLParser
 
 
-def extract(htmlstring, url='http://www.example.com/', encoding="UTF-8"):
+logger = logging.getLogger(__name__)
+
+
+def _microdata_transform(obj, schema_context, add_context=True):
```

Comment:
I think it is better to keep top-level `__init__.py` small, it should contain imports and the highest-level extract function. I'd even move `extract` function elsewhere as well.